crates/oxc_linter/src/rules/react/jsx_no_comment_textnodes.rs

```diff
@@ -70,11 +68,12 @@ impl Rule for JsxNoCommentTextnodes {
     }
 }
 
-fn control_patterns(pattern: &str) -> bool {
-    lazy_static! {
-        static ref CTL_PAT: Regex = Regex::new(r"(?m)^\s*/(/|\*)",).unwrap();
-    }
-    CTL_PAT.is_match(pattern)
+/// Returns true if the given text contains a comment pattern such as `//` or `/*`.
+fn has_comment_pattern(text: &str) -> bool {
+    text.lines().any(|line| {
+        let line = line.trim();
+        line.starts_with("//") || line.starts_with("/*")
```

Comment:
I wonder if we could get even better perf by using chars() and matching on the first two characters 🤔. We could skip double `Pattern` creation

Reply:
I tried it with `.chars()` but it didn't seem meaningfully faster at all: https://godbolt.org/z/bnE89aa6P, I'd probably prefer the `.starts_with()` method just for readability for now.